src/test/resources/GenerateTest/generate java with gen file override fails with two files/build.gradle

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright (c) 2022 Andrew Parmet
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+plugins {
+  id 'java'
+  id 'com.parmet.buf'
+}
+
+repositories {
+  mavenCentral()
+}
+
+buf {
+  generate {
+    genFileLocation = project.file("subdir/buf.gen.yaml")
+  }
+}
+
+compileJava.dependsOn 'bufGenerate'
+
+sourceSets.main.java {
+  srcDir 'build/bufbuild/generated/java'
+}
+
+dependencies {
+  implementation "com.google.protobuf:protobuf-java:$protobufVersion"
+}
```

Comment:
I think for each of these failure cases it may be better practice to minimally trigger the failure:

```groovy
plugins {
  id 'com.parmet.buf'
}

// do whatever config you need for the test here
buf {
  generate {
    genFileLocation = project.file("subdir/buf.gen.yaml")
  }
}
```

The test can run the `buf generate` task instead of the `build` task.

As a bonus none of the `.proto` or `.java` files should be needed.

Reply:
Makes sense to me. Should we remove the proto's in the GenerateWithWorkspaceTest as well? By doing so they become the same test as GenerateTest. In such case, I can move over the failure cases to not be in both test suites but just GenerateTest.

Reply:
Yes, I'd say remove the unneeded files from each concrete test class that doesn't need them. Both GenerateTest and GenerateWithWorkspaceTest should have the failure cases if they're relevant (and I think they are).

Reply:
Sounds good. I've removed the proto and buf.work.yaml files from the 2 GenerateWithWorkspaceTest failure cases